custom/abdm/models.py

```diff
@@ -0,0 +1,32 @@
+import os
+import binascii
+
+from django.db import models
+
+
+class ABDMUser(models.Model):
+    username = models.CharField(max_length=100, primary_key=True)
+    access_token = models.CharField(max_length=2000, null=True, blank=True)
+    domain = models.CharField(max_length=100, default="")
+
+    def save(self, *args, **kwargs):
+        if not self.access_token:
+            self.generate_token()
+        return super().save(*args, **kwargs)
+
+    def generate_token(self):
+        token = binascii.hexlify(os.urandom(20)).decode()
```

Comment:
I believe this is taken from `rest_framework.authtoken.models.Token` so we could directly call that?
`Token().generate_key() or you could also use `generate_key` on `HQApiKey` model.

Reply:
Changed to use from `Token` model.